CMakeLists.txt

```diff
@@ -59,19 +26,14 @@ if(DEBUG)
   set(CMAKE_CXX_FLAGS "-O0 -g")
 endif()
 
-# dl for dlopen/dlysm/dlclose
-find_library(LIBDL dl REQUIRED)
-if(LIBDL)
-  message(STATUS "Found libdl: ${LIBDL}")
-else()
-  message(FATAL_ERROR "libdl not found")
-  exit()
+if(WIN64)
+  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -DWEBGPU_BACKEND_DAWN")
 endif()
 
+include("${CMAKE_CURRENT_SOURCE_DIR}/cmake/gpu.cmake")
 
-# Build the library target (libgpu)
+message(STATUS "CMAKE_CURRENT_SOURCE_DIR: ${CMAKE_CURRENT_SOURCE_DIR}")
+message(STATUS "Include directories for wgpu: ${CMAKE_CURRENT_SOURCE_DIR}/third_party/headers")
 
-set(CMAKE_LIBRARY_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/lib)
-set(SRC_LIB gpu.h utils/shaders.h utils/array_utils.h utilslogging.h)
-add_library(gpu SHARED ${SRC_LIB})
-set_target_properties(gpu PROPERTIES LINKER_LANGUAGE CXX)
+# Add subdirectories for examples
+add_subdirectory(examples/hello_world)
```

Comment:
Maybe make a CMakeLists.txt in the examples folder and include the examples in that instead

Reply:
This makes sense, keep the root CMakeLists to building the library, I like that. Will handle this with the Makefile changes from Austin.